### DISCUSSION TITLE
Hybrid Search & Custom Scoring

### DISCUSSION THREAD

**Discussion by xiankgx** [milvus-io/milvus#22201] [Q&A and General discussion]
https://milvus.io/docs/hybridsearch.md

With ref to the above documentation, it seems that hybrid search in milvus is "a vector search with attribute filtering". I'm wondering if milvus supports custom scoring? For example, the score for a hybrid search could be based on an aggregated score of two factors: i) KNN/ANN distance/similarity score and ii) the score from conventional search (from the value of some attributes, or how well documents are matching the query (eg. BM25 score)).

For eg., for a sample/document, if the KNN distance of the sample vector to the query vector is 0.6 cosine similarity, and the download count attribute for the sample is 1.0, then the aggregated score could be 0.6 + 1.0 = 1.6.

Does milvus supports such hybrid search capabilities?

**Comment by yhmo** [comment]
Currently, Milvus doesn't support custom scoring. But it is in our feature plan: https://wiki.lfaidata.foundation/display/MIL/Feature+plans


Ranking Support | 3.0 or later | 1 | 2 | 3+ | Support complex ranking between scalar and vector score with machine learning model
-- | -- | -- | -- | -- | --


We are currently working on supporting multiple embeddings and indexes in a single collection, so after that is implemented we can move to this.
